File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
'''Class to Rectangle'''


class Rectangle:
    '''This class is create a rectangle'''

    def __init__(self, width=0, height=0):
        '''Inicialization of width and height'''
        self.height = height
        self.width = width

    @property
    def width(self):
        '''getter to retrieve the width'''
        return self.__width

    @width.setter
    def width(self, value):
        '''setter to get the size'''
        self.__width = value
        if type(value) is not int:
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")

    @property
    def height(self):
        '''getter to retrieve the height'''
        return self.__height

    @height.setter
    def height(self, value):
        '''setter to get the size'''
        self.__height = value
        if type(value) is not int:
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")

    def area(self):
        """take the square of a nimber"""
        return self.__width * self.__height

    def perimeter(self):
        if self.__width and self.__height == 0:
            return 0
        else:
            return ((self.__width + self.__height) * 2)

    def __str__(self):
        rectangle = ""
        if self.__height == 0 or self.__width == 0:
            return rectangle
        for i in range(self.__height):
            for j in range(self.__width):
                rectangle += '#'
            if j != self.__width and i != self.__height:
                if (i+1) != self.__height:
                    rectangle += '\n'
        return rectangle

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
module.exports = class Rectangle {
  constructor (w, h) {
    if (w && h <= 0) {
      return 0;
    } else if (!(w && h)) {
      return 0;
    }
    this.width = w;
    this.height = h;
  }

  print () {
    let x;
    let y;
    for (x = 0; x < this.height; x++) {
      let rect = '';
      for (y = 0; y < this.width; y++) {
        rect += 'X';
      }
      console.log(rect);
    }
  }
};
